src/custom/auth.ts

```diff
@@ -13,44 +14,88 @@ type AwsAuthLoginOptions = {
 export default class AuthClient {
 	#sdkAuthenticator: AuthenticatorFunction;
 	#apiClient: InfisicalApi;
-	#baseUrl: string;
+	#accessToken: string | undefined;
 
-	constructor(authenticator: AuthenticatorFunction, apiInstance: InfisicalApi, baseUrl: string) {
+	constructor(authenticator: AuthenticatorFunction, apiInstance: InfisicalApi, accessToken?: string) {
 		this.#sdkAuthenticator = authenticator;
 		this.#apiClient = apiInstance;
-		this.#baseUrl = baseUrl;
+		this.#accessToken = accessToken;
 	}
 
+	#renewToken = async () => {
+		try {
+			if (!this.#accessToken) {
+				throw new InfisicalError("Unable to renew access token, no access token set. Are you sure you're authenticated?");
+			}
+
+			const res = await this.#apiClient.apiV1AuthTokenRenewPost({
+				apiV1AuthTokenRenewPostRequest: {
+					accessToken: this.#accessToken
+				}
+			});
+
+			return res.data.accessToken;
+		} catch (err) {
```

Comment:
The pattern seems very redundant. Maybe you can catch things at the instance level? That way, you catch everything in one spot. Worth doing a bit of research into